src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannel.cs

```diff
@@ -2263,7 +2263,7 @@ private void CloseWaitHandle()
 
             internal class AsyncWaiter : AsyncResult, IWaiter
             {
-                private static Action<object> s_timerCallback = new Action<object>(TimerCallback);
+                private static Action<object> s_timerCallback = Fx.ThunkCallback<object>(TimerCallback);
 
```

Comment:
Can you rename the static method TimerCallback as it took me a moment to realize TimerCallback is both a local static method and the delegate definition System.Threading.TimerCallback. Either that, or change line 2281 to read:
```csharp
_timer = new Timer(new System.Threading.TimerCallback(s_timerCallback), this, timeout, TimeSpan.FromMilliseconds(-1));
```
Right now the code is confusingly ambiguous.
Also, although you don't need to change this, you could have created an overload of ThunkCallback which takes and returns a System.Threading.TimerCallback and you could avoid the new operation on line 2281. You could consider changing this line to be:
```
private static System.Threading.TimerCallback s_timerCallback = new System.Threading.TimerCallback(Fx.ThunkCallback<object>(TimerCallback));
```
That way you avoid the new operator on line 2281 on each instantiation. All suggestions, nothing required.